modules/backup.nix

```diff
@@ -1,10 +1,24 @@
 { lib
 , config
+, pkgs
 , ...
 }:
 
 let
   cfg = config.services.backup;
+
+  mkZfsPreHook = mountpoint: ''
+    DATASET="$(findmnt -nr -o -target -T "${mountpoint}")"
+    zfs snapshot -r "$DATASET@borg"
```

Comment:
Can we make this loudly fail if the snapshot already exists to avoid any potential race conditions?

Reply:
(It might already, but I'm not sure.)

Reply:
It already will.

```
 borgbackup-job-state-start[456920]: cannot create snapshot 'zroot/root/prometheus@backup': dataset already exists
 borgbackup-job-state-start[456920]: no snapshots were created
 borgbackup-job-state-start[456926]: could not find any snapshots to destroy; check snapshot names.
 systemd[1]: borgbackup-job-state.service: Main process exited, code=exited, status=1/FAILURE
 systemd[1]: borgbackup-job-state.service: Failed with result 'exit-code'.